pkg/util/ring.go

```diff
@@ -32,24 +34,15 @@ func IsInReplicationSet(r *ring.Ring, ringKey uint32, address string) (bool, err
 	return false, nil
 }
 
-// IsAssignedKey replies wether the given component instance address is in the ReplicationSet responsible for the given key or not.
+// IsAssignedKey replies wether the given instance address is in the ReplicationSet responsible for the given key or not, based on the tokens.
 //
-// The result will be defined based on the tokens assigned to each ring component.
-func IsAssignedKey(ringClient ring.ReadRing, instance HasDedicatedAddress, key string) bool {
-	bufDescs, bufHosts, bufZones := ring.MakeBuffersForGet()
+// The result will be defined based on the tokens assigned to each ring component, queried through the ring client.
+func IsAssignedKey(ringClient ring.ReadRing, instanceAddress string, key string) bool {
 	token := TokenFor(key, "" /* labels */)
-	rs, err := ringClient.Get(token, ring.WriteNoExtend, bufDescs, bufHosts, bufZones)
-	if err == nil {
-		addr := instance.GetInstanceAddr()
-		if !rs.Includes(addr) {
-			return false
-		}
+	inSet, err := IsInReplicationSet(ringClient, token, instanceAddress)
+	if err != nil {
+		level.Error(util_log.Logger).Log("msg", "error checking if key is in replicationset", "error", err, "key", key)
+		return true
```

Comment:
```suggestion
		return false
```
Let us return `false` here to avoid downloading the index for all the tenants when the ring is down. It is okay to not do query readiness for tenants when ring is down since we anyways would download their index at query time but downloading it for all users might make us run out of disk if there are too many tenants.

Reply:
Hmmm are you sure it would be better? I think true fits better here because:
- We're not exactly going to download all index for all tenants when the ring is down; what will happen instead is that if the ring is down, the `simple` mode behavior will take place, so we won't have any regressions AFAIK
- If we return false here, we'll do nothing during query readiness in case the ring is down. I'm not sure what is the best behavior though.

That said, if you still think false looks better here just lmk, I think it would work nicely too.

Reply:
In `simple` mode we do download index for all the tenants, of course if it is configured so :)

Reply:
got it, changed!